Added `load_config` function

james-mchugh opened the pull request:
Added load config function from https://github.com/kubernetes-client/python/blob/master/kubernetes/base/config/__init__.py#L24. It was pretty much a drop-in copy and paste, aside from having to update the `load_kube_config` call to be `await`ed. I would have copied over any tests as well, but I did not find any. Please let me know if you would like to build some up out or if you'd like me to make any other updates!

Fixes #249

tomplus commented on kubernetes_asyncio/config/__init__.py:
```diff
@@ -12,9 +12,35 @@
+    if "kube_config_path" in kwargs.keys() or exists(expanduser(KUBE_CONFIG_DEFAULT_LOCATION)):
+        await load_kube_config(**kwargs)
+    else:
+        print(
```
I think this print() could be annoying if you run your code as a local script and as a pod in a cluster. 
We can:
1. remove it, ;)
2. use logging.debug instead
3. print only if load_incluster_config failed.

what do you think?

james-mchugh replied:
Given my experience with the same print message from the original `kubernetes` package, I certainly agree it can be annoying lol. I just erred on the conservative side with the changes I made to the original code. What are your thoughts on using a warning instead? That way, it only prints once.

tomplus replied:
Sure, a warning will be good :+1:

tomplus reviewed (commented):
Thanks for you porting this code. It's definitely helpful in some use cases.

Could you add a simple test to be sure that the function is in place?
You can add something like this: https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/config/kube_config_test.py#L950 but using the new function. Thanks.

james-mchugh commented:
> Thanks for you porting this code. It's definitely helpful in some use cases.
> 
> Could you add a simple test to be sure that the function is in place? You can add something like this: https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/config/kube_config_test.py#L950 but using the new function. Thanks.

Can do

james-mchugh commented:
Okay, after a bit of a slow down trying to write unit tests using `unittest` again (I've been using Pytest for the past 5ish years),  I've added two unit tests. One test is for loading the kube config and the other is for loading the incluster config. I had considered adding a dedicated test file, but ultimately decided to add them to the existing files to make better use of the existing helpers they contained (I miss fixtures). 

The kube config test was fairly simple, but the incluster config test was a bit more challenging. The `load_incluster_config` function isn't super testable as is (it relied on hardcoded file paths that were difficult to monkeypatch). I updated the function to make it a bit more configurable and testable.

I also noticed that the version of `load_config` I copied from Kubernetes was a bit outdated. I'm assuming I had an old version of the`kubernetes` Python package locally that I copied the config from, so I've updated that to use the newer code as well.

tomplus reviewed (commented):
Thanks, looks good to me :+1:

Pipelines reported an error, could you take a look?

james-mchugh commented:
> Pipelines reported an error, could you take a look?

Awh my bad, I was testing using Python 3.11 and the parentheses in `with` statements [wasn't added until 3.10](https://docs.python.org/3/reference/compound_stmts.html#with:~:text=Changed%20in%20version%203.10%3A%20Support%20for%20using%20grouping%20parentheses%20to%20break%20the%20statement%20in%20multiple%20lines.). It should be good now 👍.

It may be worth adding a [tox.ini file](https://tox.wiki/en/4.16.0/index.html) in the future to make it easier to run tests across different Python versions. Let me know if that sounds like something you'd be interested in and I could perhaps put in a PR for it

tomplus commented:
I removed tox.ini (https://github.com/tomplus/kubernetes_asyncio/pull/105/), but now I agree it could be helpful.
So yes, If you want to add it again please do it - many thanks for your contribution!
